docs/api-reference/dbos-compiler.md

```diff
@@ -0,0 +1,80 @@
+For example, if the application version was `1720214256655`, all of the stored procedures generated by would be prefixed with `v1720214256655`.
+
+:::warning
+DBOS Cloud cleans up the database deployed stored procedures for older application versions via a background task.
+Do not attempt to cleanup DBOS cloud deployed stored procedure versions yourself!
+:::
+
+## Commands
+
+---
+
+### `npx dbosc deploy`
+
+**Description:**
+This command deploys the stored procedure functions from a DBOS application to the PostgreSQL database specified in the [configuration file](./configuration).
+You must deploy your stored procedures to the database before running your DBOS application.
+
+**Arguments:**
+- `tsconfigPath`: path to the DBOS application's [tsconfig.json file](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html).
+  If this argument is not provided, `dbosc` will use the tsconfig.json file from the appDirectory (if specified) or the current directory.
+
+**Parameters:**
+- `-d, --appDir <application-directory>`: The path to your application root directory.
+- `--app-version <string>` / `--no-app-version`: Overrides the `DBOS__APPVERSION` environment variable.
+  For more details, see [Stored Procedure Versioning](#stored-procedure-versioning) above.
+
+---
+
+### `npx dbosc drop`
+This command drops the DBOS application's stored procedures from the PostgreSQL database specified in the [configuration file](./configuration). 
+
+**Arguments:**
+- `tsconfigPath`: path to the DBOS application's [tsconfig.json file](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html).
+  If this argument is not provided, `dbosc` will use the tsconfig.json file from the appDirectory (if specified) or the current directory.
+
+**Parameters:**
+- `-d, --appDir <application-directory>`: The path to your application root directory.
+- `--app-version <string>` / `--no-app-version`: Overrides the `DBOS__APPVERSION` environment variable.
+  For more details, see [Stored Procedure Versioning](#stored-procedure-versioning) above.
+
+---
+
+### `npx dbosc compile`
+This command generates `create.sql` and `drop.sql` files containing the SQL commands to deploy or drop the DBOS application stored procedures. 
+This command can be useful to integrate DBOS into an environment where you can't deploy stored procedures to the database automatically with `dbosc deploy`.
+
+:::warning
+This command will overwrite existing `create.sql` and `drop.sql` files in the output directory.
+:::
+
+**Arguments:**
+- `tsconfigPath`: path to the DBOS application's [tsconfig.json file](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html).
+  If this argument is not provided, `dbosc` will use the tsconfig.json file from the current directory.
+
+**Parameters:**
+- `-o, --out <string>`: The path to generate the `create.sql` and `drop.sql` files to. Defaults to current directory if not specified.
```

Comment:
Is it the path to the directory? Or does `--out` includes the file name as well?